src/analysis/complex/cauchy_integral.lean

```diff
@@ -0,0 +1,363 @@
+/-
+Copyright (c) 2021 Yury Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury Kudryashov
+-/
+import measure_theory.measure.complex_lebesgue
+import measure_theory.integral.divergence_theorem
+import measure_theory.integral.circle_integral
+import analysis.analytic.basic
+
+/-!
+# Cauchy integral formula
+
+In this file we prove Cauchy theorem and Cauchy integral formula for integrals over circles. Most
+results are formulated for a function `f : ℂ → E` that takes values in a complex Banach space with
+second countable topology.
+
+## Main statements
+
+* `complex.integral_boundary_rect_of_has_fderiv_within_at_real_off_countable`: If a function
+  `f : ℂ → E` is *real* differentiable on a rectangle, then its integral over the boundary of this
+  rectangle is equal to the integral of `I • f' (x + y * I) 1 - f' (x + y * I) I` over the
+  rectangle, where `f' z w : E` is the derivative of `f` at `z` in the direction `w` and
+  `I = complex.I` is the imaginary unit.
+
+* `complex.integral_boundary_rect_eq_zero_of_differentiable_on_off_countable`: If a function
+  `f : ℂ → E` is *complex* differentiable on a rectangle, then its integral over the boundary of
+  this rectangle is equal to zero.
+
+* `complex.circle_integral_sub_center_inv_smul_eq_of_differentiable_on_annulus_off_countable`:
+  If a function `f : ℂ → E` is complex differentiable on an annulus `{z | r ≤ |z - c| ≤ R}`,
+  then the integrals of `(z - c)⁻¹ • f z` over the outer boundary and over the inner boundary are
+  equal.
+
+* `complex.circle_integral_sub_center_inv_smul_of_differentiable_on_off_countable_of_tendsto`,
+  `complex.circle_integral_sub_center_inv_smul_of_differentiable_on_off_countable`:
+  If a function `f : ℂ → E` is complex differentiable on a punctured closed disc
+  `{z | |z - c| ≤ R ∧ z ≠ c}` and tends to `y` as `z → c`, `z ≠ c`, then the integral of
+  `(z - c)⁻¹ • f z` over the circle `|z - c| = R` is equal to `2πiy`. In particular, if `f`
+  is differentiable on the whole closed disc, then this integral is equal to `2πif(c)`.
+
+* `complex.circle_integral_sub_inv_smul_of_differentiable_on`, **Cauchy integral formula**: if
+  `f : ℂ → E` is complex differentiable on a closed disc of radius `R`, then for any `w` in the
+  corresponding open disc the integral of `(z - w)⁻¹ • f z` over the boundary of the disc is equal
+  to `2πif(w)`.
+
+* `differentiable_on.has_fpower_series_on_ball`: If `f : ℂ → E` is complex differentiable on a
+  closed disc of positive radius, then it is analytic on the corresponding open disc, and the
+  coefficients of the power series are given by Cauchy integral formulas.
+
+* `differentiable_on.analytic_at`, `differentiable.analytic_at`: If `f : ℂ → E` is differentiable
+  on a neighborhood of a point, then it is analytic at this point. In particular, if `f : ℂ → E`
+  in differentiable on the whole `ℂ`, then it is analytic at every point `z : ℂ`.
+
```

Comment:
Could you sketch in the file-level docstring the strategy you are using to prove these theorems (and also emphasize that you are only assuming differentiability, not continuity of the derivative, contrary to more elementary treatments of the theory)? Also, since you are not following the standard textbook path, it may help the reader to see where something nontrivial is happening (in the divergence theorem with super-weak assumptions)